libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Conway/Instances.hs

```diff
@@ -316,7 +328,7 @@ type BabbageTxOutTypes era =
    , StrictMaybe (Script era)
    ]
 instance (Era era, Val (Value era)) => HasSimpleRep (BabbageTxOut era) where
-  type SimpleRep (BabbageTxOut era) = SOP '["BabbageTxOut" ::: BabbageTxOutTypes era]
+  type TheSop (BabbageTxOut era) = '["BabbageTxOut" ::: BabbageTxOutTypes era]
   toSimpleRep (BabbageTxOut addr val dat msc) =
```

Comment:
What is the difference between 'SimplRep' and 'TheSop'?

Reply:
It's an annoying technical detail - `SimpleRep a = SOP (TheSop a)` and `TheSop = SOPOf (Rep a)` in the generic case - this is what lets you write just `instance ... => HasSimpleRep A` when you don't want a specific instance. When you _do_ want a specific instance you need to define `TheSop` rather than `SimpleRep` if you want `sel`, `con`, and `is/onCon` to work properly. It's something I'm planning on fixing.